source/common/http/http1/parser_impl.cc

```diff
@@ -0,0 +1,226 @@
+#include "common/http/http1/parser_impl.h"
+
+#include <llhttp.h>
+
+#include <iostream>
+
+#include "common/common/assert.h"
+#include "common/http/http1/parser.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http1 {
+namespace {
+ParserStatus intToStatus(int rc) {
+  // See
+  // https://github.com/nodejs/http-parser/blob/5c5b3ac62662736de9e71640a8dc16da45b32503/http_parser.h#L72.
+  switch (rc) {
+  case -1:
+    return ParserStatus::Error;
+  case 0:
+    return ParserStatus::Success;
+  case 1:
+    return ParserStatus::NoBody;
+  case 2:
+    return ParserStatus::NoBodyData;
+  case 21:
+    return ParserStatus::Paused;
+  default:
+    return ParserStatus::Unknown;
+  }
+}
+} // namespace
+
+class HttpParserImpl::Impl {
+public:
+  Impl(llhttp_type_t type) {
+    llhttp_init(&parser_, type, &settings_);
+    llhttp_set_lenient_chunked_length(&parser_, 1);
+    llhttp_set_lenient_headers(&parser_, 1);
```

Comment:
Well, maybe. We have our own code for header validation. It would change error code/messages. But since this feature has to be turned on and deprecated maybe this is a great time to do this and remove some codec code.

Will do. And leaving comment in codec code that we will be able to remove this after removing llhttp.